slim.Dockerfile

```diff
@@ -3,7 +3,8 @@ FROM python:3.7-alpine as builder
 WORKDIR /girder
 RUN apk add --no-cache git gcc musl-dev libffi-dev openssl-dev
 COPY . ./
-RUN pip install --no-cache-dir .
+# Both PYTHONDONTWRITEBYTECODE and --no-compile are necessary to avoid creating .pyc files
+RUN PYTHONDONTWRITEBYTECODE=1 pip install --no-cache-dir --no-compile .
```

Comment:
Since this is in the build stage, does it matter?  Do these `.pyc` files end up in site-packages or elsewhere?

Reply:
How big are the bytecode files anyway? If they're not too big they're probably worth keeping around for the performance boost.

Reply:
Yes, they go inside `__pycache__` directories scattered throughout `site-packages`, and will be copied along with `site-packages` otherwise.

Reply:
Unless we prevent it (via an environment variable or python flag), they get regenerated as needed and only have a cost on first use of a module.

Reply:
It reduces the build size by ~15MB, FWIW.

Reply:
:+1: from me, but I don't know python mechanics well enough to make a good argument either way.

Reply:
We can leave them off then. I guess the point of this image is to be small, not necessarily performant.